modules/standard/AutoGpu.chpl

```diff
@@ -47,6 +103,32 @@ module AutoGpu {
     __primitive("chpl_assert_on_gpu");
   }
 
+  /*
+   This attribute can be applied to loops and variable declarations to specify
+   the GPU block size to use when executing the loop on the GPU.
+
+   When applied to loops, this attribute specifies the block size for the
+   loop itself. When applied to variables, this attribute specifies the block
+   size for all promoted expressions and loop expressions (e.g., foreach
+   expressions) instead.
+
+   The ``counter`` parameter is automatically generated by the compiler,
+   and should not be specified by the user.
+
+   .. code-block:: chapel
+
+     // loop version
+     @gpu.blockSize(64)
+     foreach i in 1..128 { /* ... */ }
+
+     // variable version
+     @gpu.blockSize(64)
+     var A = (foreach i in 1..128 do i*i) + 1;
+  */
+  pragma "docs only"
+  @chpldoc.attributeSignature("gpu.blockSize")
+  inline proc chpl__gpuBlockSizeAttr(arg: integral) {}
```

Comment:
Why does this specify a different number of arguments than the version below it?

Reply:
The `counter` argument is auto-inserted by the compiler to "tag" instances of the GPU primitive. They end up duplicated in some cases, and that tag is used as a "cache key" to detect when two copies of the primitives originated from the same call. This is a compiler detail, so I hide it here.